remote_attestation/rust/src/tests/crypto.rs

```diff
@@ -0,0 +1,224 @@
+];
+const CLIENT_KEY_AGREEMENT_PUBLIC_KEY: [u8; KEY_AGREEMENT_ALGORITHM_KEY_LENGTH] = [
+    75, 130, 163, 244, 38, 189, 249, 222, 43, 127, 116, 150, 233, 190, 243, 236, 19, 237, 74, 108,
+    9, 122, 105, 223, 34, 125, 125, 242, 239, 66, 78, 95,
+];
+const SERVER_ENCRYPTION_KEY: [u8; AEAD_ALGORITHM_KEY_LENGTH] = [
+    148, 3, 161, 94, 41, 6, 250, 223, 92, 22, 86, 91, 150, 171, 248, 51, 221, 90, 230, 90, 151, 55,
+    59, 108, 21, 54, 33, 61, 219, 140, 229, 59,
+];
+const CLIENT_ENCRYPTION_KEY: [u8; AEAD_ALGORITHM_KEY_LENGTH] = [
+    48, 90, 139, 218, 156, 196, 174, 129, 15, 178, 216, 181, 106, 232, 160, 188, 136, 103, 87, 120,
+    231, 20, 78, 129, 237, 225, 173, 25, 179, 152, 139, 82,
+];
+const SIGNING_PUBLIC_KEY: [u8; SIGNING_ALGORITHM_KEY_LENGTH] = [
+    4, 51, 234, 203, 59, 199, 24, 233, 70, 68, 246, 116, 56, 113, 159, 19, 252, 238, 243, 127, 133,
+    116, 177, 172, 54, 42, 233, 123, 233, 25, 181, 153, 133, 63, 45, 191, 156, 46, 102, 156, 93,
+    183, 74, 48, 189, 37, 49, 50, 66, 90, 61, 100, 2, 81, 180, 225, 13, 253, 220, 7, 54, 127, 13,
+    131, 85,
+];
+
+const DATA: [u8; 10] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+// [`DATA`] encrypted with [`CLIENT_ENCRYPTION_KEY`],
+const ENCRYPTED_DATA: [u8; 26] = [
+    255, 242, 249, 3, 114, 107, 147, 122, 38, 153, 12, 33, 62, 56, 172, 90, 234, 207, 50, 219, 22,
+    212, 169, 40, 113, 28,
+];
+const INVALID_ENCRYPTED_DATA: [u8; 26] = [0; 26];
+const ENCRYPTED_DATA_NONCE: [u8; NONCE_LENGTH] =
+    [106, 176, 114, 112, 226, 142, 211, 123, 95, 187, 120, 206];
+const INVALID_ENCRYPTED_DATA_NONCE: [u8; NONCE_LENGTH] = [0; NONCE_LENGTH];
+const DATA_SIGNATURE: [u8; SIGNATURE_LENGTH] = [
+    14, 133, 10, 201, 218, 60, 123, 206, 59, 25, 101, 172, 89, 65, 201, 197, 92, 249, 247, 99, 133,
+    8, 217, 94, 125, 160, 31, 141, 61, 197, 119, 2, 41, 79, 195, 23, 105, 0, 50, 253, 111, 103, 19,
+    188, 169, 243, 11, 157, 49, 155, 154, 249, 14, 29, 183, 167, 0, 82, 74, 74, 221, 214, 250, 222,
+];
+const INVALID_DATA_SIGNATURE: [u8; SIGNATURE_LENGTH] = [0; SIGNATURE_LENGTH];
+const DATA_SHA256_HASH: [u8; SHA256_HASH_LENGTH] = [
+    31, 130, 90, 162, 240, 2, 14, 247, 207, 145, 223, 163, 13, 164, 102, 141, 121, 28, 93, 72, 36,
+    252, 142, 65, 53, 75, 137, 236, 5, 121, 90, 179,
+];
+
+#[test]
+fn test_decrypt() {
+    let mut encryptor = AeadEncryptor::new(SERVER_ENCRYPTION_KEY, CLIENT_ENCRYPTION_KEY);
+    let result = encryptor.decrypt(&EncryptedData::new(
+        ENCRYPTED_DATA_NONCE,
+        ENCRYPTED_DATA.to_vec(),
+    ));
+    assert_eq!(result.unwrap(), DATA);
+
+    let result = encryptor.decrypt(&EncryptedData::new(
+        INVALID_ENCRYPTED_DATA_NONCE,
+        ENCRYPTED_DATA.to_vec(),
+    ));
+    assert!(result.is_err());
+    let result = encryptor.decrypt(&EncryptedData::new(
+        ENCRYPTED_DATA_NONCE,
+        INVALID_ENCRYPTED_DATA.to_vec(),
+    ));
+    assert!(result.is_err());
+}
+
+#[quickcheck]
+fn test_encrypt(data: Vec<u8>) -> bool {
```

Comment:
Should this be called round_trip? ref https://github.com/project-oak/oak/pull/2286#discussion_r713719070